<workflow-app xmlns="uri:oozie:workflow:0.4" name="metadataextraction_cache_manual_lock_manager">
	
	<parameters>
		<property>
			<name>zk_session_timeout</name>
			<value>60000</value>
			<description>zookeeper session timeout when handling locks (expressed in milliseconds)</description>
		</property>
		<property>
			<name>cache_location</name>
			<description>default cache location stored in HDFS</description>
		</property>
		<property>
			<name>mode</name>
			<value>release</value>
			<description>lock managing mode: obtain or release values are supported</description>
		</property>
	</parameters>

    <global>
        <job-tracker>${jobTracker}</job-tracker>
        <name-node>${nameNode}</name-node>
        <configuration>
            <property>
                <name>mapreduce.job.queuename</name>
                <value>${queueName}</value>
            </property>
            <property>
                <name>oozie.launcher.mapred.job.queue.name</name>
                <value>${oozieLauncherQueueName}</value>
            </property>
        </configuration>
    </global>
    
    
	<start to="manage-lock-and-end" />

    <action name="manage-lock-and-end">
        <java>
			<main-class>eu.dnetlib.iis.common.java.ProcessWrapper</main-class>
			<arg>eu.dnetlib.iis.common.lock.LockManagingProcess</arg>
            <arg>-Pzk_session_timeout=${zk_session_timeout}</arg>
            <arg>-Pnode_id=${cache_location}</arg>
            <arg>-Pmode=${mode}</arg>
        </java>
        <ok to="end"/>
        <error to="fail"/>
    </action>
    
	<kill name="fail">
		<message>Unfortunately, the process failed -- error message:
			[${wf:errorMessage(wf:lastErrorNode())}]</message>
	</kill>
	<end name="end" />
</workflow-app>
